include/SDL3/SDL_stdinc.h

```diff
@@ -498,6 +498,9 @@ extern DECLSPEC int SDLCALL SDL_setenv(const char *name, const char *value, int
 extern DECLSPEC void SDLCALL SDL_qsort(void *base, size_t nmemb, size_t size, int (SDLCALL *compare) (const void *, const void *));
 extern DECLSPEC void * SDLCALL SDL_bsearch(const void *key, const void *base, size_t nmemb, size_t size, int (SDLCALL *compare) (const void *, const void *));
 
+extern DECLSPEC void SDLCALL SDL_qsort_r(void *base, size_t nmemb, size_t size, int (SDLCALL *compare) (const void *, const void *, void *), void *userdata);
+extern DECLSPEC void * SDLCALL SDL_bsearch_r(const void *key, const void *base, size_t nmemb, size_t size, int (SDLCALL *compare) (const void *, const void *, void *), void *userdata);
```

Comment:
I just want to bring up the point that most (all?) SDL callback function have `void *userdata` as their first argument (similar to the Macos definition).

Reply:
Is there a prior art reason not to do this? In general we want the callback to flow into the C++ `this` calling convention.

Reply:
This would be the first callback function pointer accepting a userdata pointer in `SDL_stdinc.h`.

None of the new functions are [defined in posix](https://pubs.opengroup.org/onlinepubs/9699919799/functions/contents.html), so I think we are free to do whatever we want. It's the wild west.
Because the implementation of Macos and glibc differ, it is unlikely these will ever be standardized.

Reply:
I also could not find any posix functions that accept a userdata pointer. Or I looked in the wrong place.

Reply:
> Is there a prior art reason not to do this? In general we want the callback to flow into the C++ `this` calling convention.

It follows glibc, maybe because of this? https://sourceware.org/legacy-ml/libc-alpha/2008-12/msg00007.html

Reply:
I did it to match glibc, and then realized there is no standard at all for qsort_r. I can change it to match SDL standards.

Reply:
Switched the callback argument order in 01ef5226cb09ec096ba5b4dfa8a68a21c5e4ac7c.